app/services/courses/assign_subjects_service.rb

```diff
@@ -0,0 +1,52 @@
+module Courses
+  class AssignSubjectsService
+    include ServicePattern
+
+    attr_reader :course, :subject_ids
+
+    def initialize(course:, subject_ids:)
+      @course = course
+      @subject_ids = subject_ids || []
+    end
+
+    def call
+      course.errors.add(:subjects, :duplicate) if request_has_duplicate_subject_ids?
+
+      update_subjects
+
+      course.name = course.generate_name
+      course
+    end
+
+    def subjects
```

Comment:
Do all the methods below `call` need to be private?

Reply:
done